go/database/mpt/rlp/rlp.go

```diff
@@ -253,3 +364,45 @@ func (i BigInt) getEncodedLength() int {
 	length := ((bitlen + 7) & -8) >> 3
 	return getEncodedLengthLength(length) + length
 }
+
+// readNumber interprets the first 'slen' bytes of 'b' as a big-endian integer.
+// It returns the resulting integer and any error encountered.
+//
+// The function expects 'slen' to be the size of the integer to be read from 'b'.
+// If 'slen' is greater than the length of 'b', an error is returned.
+//
+// The function supports reading integers of up to 8 bytes (64 bits).
+// The integer is constructed by shifting and OR-ing the bytes from 'b'.
+//
+// Parameters:
+// b:    The byte slice from which to read the integer.
+// slen: The size of the integer to be read from 'b'.
+//
+// Returns:
+// The integer read from 'b' and an error if 'slen' is greater than the length of 'b'.
+func readNumber(b []byte, slen byte) (uint64, error) {
+	if int(slen) > len(b) {
+		return 0, fmt.Errorf("expected %d bytes, got: %d", slen, len(b))
+	}
+	var s uint64
+	switch slen {
+	case 1:
+		s = uint64(b[0])
+	case 2:
+		s = uint64(b[0])<<8 | uint64(b[1])
+	case 3:
+		s = uint64(b[0])<<16 | uint64(b[1])<<8 | uint64(b[2])
+	case 4:
+		s = uint64(b[0])<<24 | uint64(b[1])<<16 | uint64(b[2])<<8 | uint64(b[3])
+	case 5:
+		s = uint64(b[0])<<32 | uint64(b[1])<<24 | uint64(b[2])<<16 | uint64(b[3])<<8 | uint64(b[4])
+	case 6:
+		s = uint64(b[0])<<40 | uint64(b[1])<<32 | uint64(b[2])<<24 | uint64(b[3])<<16 | uint64(b[4])<<8 | uint64(b[5])
+	case 7:
+		s = uint64(b[0])<<48 | uint64(b[1])<<40 | uint64(b[2])<<32 | uint64(b[3])<<24 | uint64(b[4])<<16 | uint64(b[5])<<8 | uint64(b[6])
+	case 8:
+		s = uint64(b[0])<<56 | uint64(b[1])<<48 | uint64(b[2])<<40 | uint64(b[3])<<32 | uint64(b[4])<<24 | uint64(b[5])<<16 | uint64(b[6])<<8 | uint64(b[7])
+	}
+
+	return s, nil
```

Comment:
Maybe shorten this to
```suggestion
	var s uint64
	for i := byte(0); i < slen; i++ {
		s = s<<8 | uint64(b[i])
	}
	return s, nil
```